### PR TITLE
Add sample for data refining using open ai 

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,19 +5,19 @@ This repository contains various code samples for ETL tasks written in Ballerina
 Each subdirectory contains relevant code and instructions for the corresponding task. Explore the subdirectories for detailed code samples and documentation.
 
 #### Data Extraction (E)
-- [Extract Data from Database](extract-data-from-database): Sample code for extracting customer orders from database.
-- [Extract Information From Unstructured Data](extract-information-from-unstructured-data): Sample code for extracting required information from unstructured data using AI.
-- [Extract Data from GitHub](extract-data-from-github): Sample code samples for extracting data from GitHub.
+- [Extract data from database](extract-data-from-database): Sample code for extracting customer orders from database.
+- [Extract information from unstructured data](extract-information-from-unstructured-data): Sample code for extracting required information from unstructured data using AI.
+- [Extract data from gitHub](extract-data-from-github): Sample code samples for extracting data from GitHub.
```
Comment (pcnfernando) [inline README.md]: Having 'GitHub' is fine

File: README.md [hunk 2]
```diff
@@ -5,19 +5,19 @@ This repository contains various code samples for ETL tasks written in Ballerina
...
 #### Data Transformation (T)
-- [Remove Duplicates](remove-duplicates): Sample code for deduplication.
-- [Remove Approximate Duplicates](remove-approximate-duplicates): Sample code for approximate deduplication.
-- [Error Correction](error-correction): Sample code for error correction in data.
-- [Clean and Standardize Data](clean-and-standardize-data): Sample code for cleaning and standardizing data.
-- [Data Mapper](data-mapper): Sample code for data mapping between different structures.
-- [Enrich Data](enrich-data): Sample code to enrich data.
+- [Remove duplicates](remove-duplicates): Sample code for deduplication.
+- [Remove approximate duplicates](remove-approximate-duplicates): Sample code for approximate deduplication.
+- [Refine data using ai](refine-data-using-ai): Sample code for error correction in data.
```
Comment (pcnfernando) [inline README.md]: AI should also be fine

File: README.md [hunk 3]
```diff
@@ -5,19 +5,19 @@ This repository contains various code samples for ETL tasks written in Ballerina
...
+- [Data mapper](data-mapper): Sample code for data mapping between different structures.
+- [Enrich data](enrich-data): Sample code to enrich data.
 
 #### Data Loading (L)
-- [Load Data to BigQuery](load-data-to-bigquery): Sample code for loading data into BigQuery.
-- [Kafka Database Ingestion](kafka-db-ingestion): Sample code for ingesting data from Kafka into a database.
-- [Load Data to Google Sheets](load-data-to-gsheet): Sample code for loading data into Google Sheets.
+- [Load data to BigQuery](load-data-to-bigquery): Sample code for loading data into BigQuery.
+- [Kafka database ingestion](kafka-db-ingestion): Sample code for ingesting data from Kafka into a database.
+- [Load data to google sheets](load-data-to-gsheet): Sample code for loading data into Google Sheets.
```
Comment (pcnfernando) [inline README.md]: Same for Google Sheets. Can you double check how other usecases handles such scenarios

File: README.md (hunk 1 above)
Comment (LakshanWeerasinghe) [reply README.md]: updated in [commit](https://github.com/ballerina-guides/etl-samples/pull/15/commits/efa507c9dc10531c031b38c0f5760f42265cad32).

File: README.md (hunk 2 above)
Comment (LakshanWeerasinghe) [reply README.md]: updated in [commit](https://github.com/ballerina-guides/etl-samples/pull/15/commits/efa507c9dc10531c031b38c0f5760f42265cad32).

File: README.md (hunk 3 above)
Comment (LakshanWeerasinghe) [reply README.md]: updated in [commit](https://github.com/ballerina-guides/etl-samples/pull/15/commits/efa507c9dc10531c031b38c0f5760f42265cad32).

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**